data/avro-schema/src/main/resources/avro/net/corda/data/crypto/wire/ops/key/rotation/IndividualKeyRotationRequest.avsc

```diff
@@ -16,18 +16,23 @@
     },
     {
       "name": "oldParentKeyAlias",
-      "type": "string",
-      "doc": "The key alias whose protected content will be re-wrapped with a new key."
+      "type": ["null", "string"],
+      "doc": "For unmanaged key rotation. The key alias whose protected content will be re-wrapped with a new key."
     },
     {
       "name": "newParentKeyAlias",
-      "type": "string",
-      "doc": "The new wrapping key which oldParentKeyAlias' content will be re-wrapped with."
+      "type": ["null", "string"],
+      "doc": "For unmanaged key rotation. The new wrapping key which oldParentKeyAlias' content will be re-wrapped with."
```

Comment:
What hapepns if newParentKeyAlias is null? I guess we use the crypto service default for unmanaged keys, and make up a new wrapping key for managed but let's say so.

Reply:
If it's null for unmanaged keys that's an error and it wont rotate. For managed keys you never need a new alias, because you are always rotating to a key with the same alias and a higher generation.